index.js

```diff
@@ -194,6 +235,32 @@ class DetailsDialogElement extends HTMLElement {
     if (!details) return
     toggleDetails(details, open)
   }
+
+  static get observedAttributes() {
+    return ['src', 'preload']
+  }
+
+  attributeChangedCallback(attribute: string) {
+    const details = this.parentElement
+    if (!details) return
+    const state = initialized.get(this)
+    if (!state) return
+
+    if (attribute === 'src') {
+      state.loaded = false
```

Comment:
I think we should return if loaded has already been set to true so we only attempt a single load. That will handle the case where the include-fragment is replaced with server content that contains another include-fragment. The dialog shouldn't affect that second fragment with `src` attribute changes.

Reply:
> is in the loaded state after replacing itself with the server content.

I've removed the loaded state. However, we'd need the `loaded` state if 

> The dialog shouldn't affect that second fragment with src attribute changes.

But I wonder why you think it shouldn't affect the second fragment if the src was intentionally changed? This could be a legitimate use case:

```html
<details-dialog src="?page=1">
  <include-fragment></include-fragment>
</details-dialog>
```

Where `?page=1` returns:

```html
<include-fragment>
  page 1 content
  <button data-next-page="?page=2">next page</button>
</include-fragment>
```

With:

```js
button.addEventListener('click', () => {
  dialog.src = button.getAttribute('data-next-page')
})
```

Reply:
Oh second/third thought I see how in that case src should just be set on the new include fragment 🤦🏻‍♀️ and changing dialog’s src wouldn’t load the second page anyways. 

That said, is it worth keeping a loaded state to prevent a second load on a second include fragment element?